pkg/reconciler/integration/defaults_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/google/kf/pkg/kf/testutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+)
+
+// DefaultTest defines validation for defaulting webhooks. The Resource will
+// be created and all paths will be tested for equality.
+type DefaultTest struct {
+	Path              string                 `json:"-"`
+	TestEqualityPaths []string               `json:"testEquality"`
+	Resource          map[string]interface{} `json:"resource"`
+}
+
+func (dt *DefaultTest) GetUnstructured() *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: dt.Resource,
+	}
+}
+
+func (dt *DefaultTest) GetGvr() schema.GroupVersionResource {
+	obj := dt.GetUnstructured()
+	gvk := obj.GroupVersionKind()
+
+	return schema.GroupVersionResource{
+		Group:   gvk.Group,
+		Version: gvk.Version,
+
+		// We can autosidscover this in the future from the API Server like
```

Comment:
Should we add a TODO w/ an issue?

Reply:
I don't think it's worth tracking this one. Our client generator uses the same logic, and so far none of the resources have deviated from it.

Reply:
```suggestion
		// We can autodiscover this in the future from the API Server like
```